### DISCUSSION TITLE
Full device encryption - SSH in unencrypted boot partition

### DISCUSSION THREAD

**Discussion by Lotechus** [1000001101000/Debian_on_Buffalo#175] [Q&A]
I have used the installer to encrypt everything except the /boot partition. Since the device is headless, is there a way to have SSH start in the boot partition so I can enter the password to decrypt the rest of the device?  How would I do this please?  I can either use the installer shell or remove the drive to another computer make the changes. Suggestions please.

**Comment by 1000001101000** [comment]
Good to hear from you!

I haven't looked very closely into doing something like that but am aware there is a package called `dropbear-initramfs` for doing what you are describing. 

Actually it would be really helpful you could throw a short guide together for setting that up if you get a chance to try it out. If it works out for your encryption use case it might also help provide a recovery option for folks who mess up their fstab or other issues that prevent a device from booting fully.

**Reply by Lotechus** [reply]
Thanks!  I did look into dropbear-initramfs and it seems like the correct software to do this, however it needs to be done from a already booted system from what I can tell.  The problem is there is no way to boot the system headless if I encrypt during install.  The USB port does not seem to initialize a keyboard and there is no place for a monitor.  

I'm not a debian pro by far - I'm hoping to be able to use apt either directly after install in the busybox terminal (before rebooting) to modify the boot partition, but I haven't been able to find information related to this.  

The alternative seems to be working on the drive in a x86 or x64 debian machine with the drives mounted externally - however not sure how to install something to an external boot root filesystem and then modify the bootloader.

I'm hoping someone can point me in the right direction... I feel like it can be done, but I just don't know the correct direction to 
look.

If I figure this out I will definitely post a short guide.

Thanks again.

**Comment by 1000001101000** [comment]
I would think you could boot into the installer and access the system via chroot and then setup the dropbear stuff and reboot.

Roughly speaking, I would think the recovery process would look like this:

- connect drive to a PC and mount /boot
- replace the *.buffalo files found there with the debian installer files for your device
- boot into the installer and take the install process to the partitioning screen
- select "go back" and open a terminal 
- run whatever commands are required to decrypt and mount your rootfs on /mnt
- mount your /boot as /mnt/boot.
- mount sys and proc under /mnt, bind mount /dev to /mnt/dev
- chroot into /mnt to access the system.
- run through the install process for the initramfs-dropbear stuff.
- generate new *.buffalo files with `update-initramfs -u`
- exit the chroot
- reboot

**Reply by Lotechus** [reply]
Hello again,
Finally had a chance to try this.  I made it completely through without success.  Would you mind providing some more guidance, I think I misunderstood a few things.

- I've decrypted the partition, ran lvscan and lvchange -ay to mark them as active for mounting
- I've mounted the decrypted root partition at /mnt and the unencrypted boot partition at /mnt/boot
- in /mnt/sys and /mnt/proc both directories both empty.  How would I mount sys and proc under /mnt and from what source if the decrypted root is already mounted there?  
- I'm also unsure about the bind mount of /dev to /mnt/dev.  Where is the source for /dev?

As I mentioned above I finished the other steps but it didn't work on reboot.  I'm willing to keep trying and document for tutorial purposes.  It's a fun but frustrating learning experience.

Thanks in advance :)

**Reply by Lotechus** [reply]
Ok think I figured that part out.
mount -t proc proc /mnt/proc
mount -t sysfs sys /mnt/sys
mount -o bind /dev /mnt/dev

I've mounted my separate tmp and var partitions to /mnt/tmp /mnt/var
I chroot into /mnt
installed dropbear, did the configuration but when I run update-initramfs -u 
I get; Nothing to do, exiting

And now I'm stuck again.  This is an LVM inside of an encrypted partition - although I'm not sure that matters as I'm just trying to get SSH into the unencrypted boot partition.

Does anyone have any ideas why update-initramfs says there is nothing to do?

**Reply by 1000001101000** [reply]
I think update-initramfs -u looks at all the /boot/vmlinuz-* options and build the initrd for the highest one. My guess would be that it isn’t finding any

**Reply by Lotechus** [reply]
Seems like it isn't. As available versions: (blank).  Not sure how to fix this.  Boot just has the 2 .buffalo files.  I've tried creating a copy of initrd.buffalo as initrd.img without luck.
